hw_3/middleware/validation.js

```diff
@@ -1,27 +1,25 @@
 import { errorResponse, userSchema, groupSchema } from '../validation';
 import { VALIDATION_OPTIONS } from '../constants/validation';
+import { ValidationError } from '../exceptions';
 
 export const validateUserInfo = (req, res, next) => {
     const userInfo = req.body;
     const { error } = userSchema.validate(userInfo, VALIDATION_OPTIONS);
 
     if (error) {
-        return res.status(400).json(errorResponse(error.details));
+        throw new ValidationError(errorResponse(error.details));
     }
 
-    next();
+    next(error);
```

Comment:
Looks like you don't need to pass the error to the next function. If the error is not falsy value, the error will be thrown.

Reply:
done